components/AuthorizedPage/AuthorizedPage.tsx

```diff
@@ -0,0 +1,57 @@
+import { ReactNode, useEffect, useMemo, useState } from "react";
+import useApp from "../../hooks/useApp";
+import jwt from "jsonwebtoken";
+import { isInIframe } from "../../utils/misc";
+import { DashboardTokenPayload } from "../../lib/middlewares";
+import AccessWarning from "../AccessWarning/AccessWarning";
+import LoadingPage from "../LoadingPage/LoadingPage";
+
+type AuthorizedPageProps = {
+  children: ReactNode;
+};
+
+const AuthorizedPage = ({ children }: AuthorizedPageProps) => {
+  const app = useApp()?.getState();
+
+  // Next and React 18 have issues with hydatation
+  // GH Issue: https://github.com/vercel/next.js/discussions/35773
+  // To be fixed once a proper solution is available
+  const [mounted, setMounted] = useState(false);
+  useEffect(() => setMounted(true), []);
+
+  const tokenClaims = useMemo(() => {
+    try {
+      return jwt.decode(app?.token as string);
+    } catch (e) {
+      console.error(e);
+      return null;
+    }
+  }, [app]);
+
+  const isIframe = useMemo(() => isInIframe(), []);
+  const isTokenValid =
+    tokenClaims && (tokenClaims as DashboardTokenPayload).iss === app?.domain;
```

Comment:
This is logic specific for tokens. We should move it to another file, related to token, expose some methods or a class, that will keep token-specific operations in single place. Component, responsible of UI, isn't perfect place for operations like this.

Also moving to another file and exposing method will:
1. Make tests easier to write (please do)
2. Make it easier to move to app-sdk (maybe we should - cc @krzysztofwolski @magul )